iaso/api/profiles.py

```diff
@@ -407,13 +407,23 @@ def update_user_profile(self, request, pk):
         return profile
 
     @staticmethod
-    def update_permissions(user, request):
+    def update_permissions(self, user, request):
         user.user_permissions.clear()
+        current_account = user.iaso_profile.account
+
         for permission_codename in request.data.get("user_permissions", []):
-            CustomPermissionSupport.assert_right_to_assign(request.user, permission_codename)
-            user.user_permissions.add(get_object_or_404(Permission, codename=permission_codename))
+            if permission_codename in self.module_permissions(current_account):
```

Comment:
Oh, this must have an impact on performance, you are calling module_permissions in a loop, and I fear that it means that you are going to do the same requests for eache persmission in the request data.